### PR TITLE
#2: Push integration prototype

### PR DESCRIPTION
**Pull request by mdesimone-viafoura**
- Add `dotEnv` for environment variables
- OneSignal (Mobile Push Provider) working prototype.
- Add utils to build string from notification object
- Improve models

### CODE REVIEW

File: notification-stream/src/routes/index.ts [hunk 1]
```diff
@@ -3,8 +3,15 @@ const express = require('express')
 const router = express.Router();
 
 import { Request, Response } from 'express'
+import { IRequest } from '../types';
+import { createNotification } from '../providers';
 
 router.post('/vf-webhook', (req: Request, res: Response) => {
+    const request = req.body as IRequest;
```
Comment (mdesimone-viafoura) [inline notification-stream/src/routes/index.ts:10]: still need to add error handling

File: notification-stream/package.json [hunk 2]
```diff
@@ -10,6 +10,8 @@
     "author": "Viafoura",
```
Comment (viaFarhan) [inline notification-stream/package.json:13]: can we add "engines" so we are clear what version(s) of Node this is compatible with?

File: notification-stream/src/providers/pushly.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const createPushlyNotification = ((message: string, vfUserId ? : string) => {
+
+
+});
```
Comment (viaFarhan) [inline notification-stream/src/providers/pushly.ts:4]: Also can you update your linter settings to have an empty line at the end of js/ts files?

File: notification-stream/src/types/index.ts [hunk 4]
```diff
@@ -35,11 +35,25 @@ export interface IUserExternalId {
...
+    description?: string;
+    title?: string;
     content: IPayloadContent;
     target_content_uuid: string;
 }
 
 export interface IPayloadContent {
     uuid: string;
-    body: string;
+    body?: string;
```
Comment (viaFarhan) [inline notification-stream/src/types/index.ts:47]: Are there scenarios where there is no body?

**Review by viaFarhan — APPROVED** [review]

File: notification-stream/src/types/index.ts (hunk 4 above)
Comment (mdesimone-viafoura) [reply notification-stream/src/types/index.ts:47]: it's optional in the spec. but yeah sounds weird.

**Pull request merged**